board/boards/board_functions.h

```diff
@@ -0,0 +1,25 @@
+// ///// Board functions ///// //
+// TODO: Make these config options in the board struct
```

Comment:
Let's not put this in a separate file, but rather add these as boolean fields in the board struct, makes way more sense imo

Reply:
`board_has_obd` and `board_has_relay` is kinda already covered by the `.has_harness` in the harness config. 
I think we can also add `.has_rtc` , `.has_lin` and `.has_gps` in the `board` struct.
Maybe the `has_gmlan` should be something like `.has_onboard_gmlan`?

Reply:
Kept .has_obd (as we discussed on Slack), removed .has_relay as it was not even used, everything else moved to struct.